crates/router/src/routes/gsm.rs

```diff
@@ -0,0 +1,93 @@
+use actix_web::{web, HttpRequest, Responder};
+use api_models::gsm as gsm_api_types;
+use router_env::{instrument, tracing, Flow};
+
+use super::app::AppState;
+use crate::{
+    core::{api_locking, gsm},
+    services::{api, authentication as auth},
+};
+
+#[instrument(skip_all, fields(flow = ?Flow::GsmRuleCreate))]
+pub async fn create_gsm_rule(
+    state: web::Data<AppState>,
+    req: HttpRequest,
+    json_payload: web::Json<gsm_api_types::GsmCreateRequest>,
+) -> impl Responder {
+    let payload = json_payload.into_inner();
+
+    let flow = Flow::GsmRuleCreate;
+    Box::pin(api::server_wrap(
+        flow,
+        state.clone(),
+        &req,
+        payload,
+        |state, _, payload| gsm::create_gsm_rule(state, payload),
+        &auth::AdminApiAuth,
+        api_locking::LockAction::NotApplicable,
+    ))
+    .await
+}
+
+#[instrument(skip_all, fields(flow = ?Flow::GsmRuleRetrieve))]
+pub async fn get_gsm_rule(
+    state: web::Data<AppState>,
+    req: HttpRequest,
+    json_payload: web::Json<gsm_api_types::GsmRetrieveRequest>,
```

Comment:
why is this a payload? do we not have a gsm identifier through which the gsm rule can be fetched?

Reply:
We need to query GSM with connector, flow, sub_flow, code and message